chore!: Align options per service (#3559)
SAP/cloud-sdk-js: V3-Upgrade-Guide.md

The `service-mapping.json` file is now `options-per-service.json`.
The file will not be generated by default in the input directory, as in the previous versions.

> You **must** set  the `optionsPerService` to have the generator consider the options. 

Reviewer: ```suggestion
If the generator should consider options, it is **mandatory** to set the `optionsPerService` argument.
```

You can set the value to either:

- The file path containing the options per service (e.g. `options.json`).
  If the file does not exist, it will be created and initialized.
  If the file exists, missing or partial service options will be added with the default values. 

> - The directory from which the file is read/created (e.g. `someDir`. This will read/create a file named `options-per-service.json` in `someDir`)

Reviewer: how does the generator distinguish directories and files? do files have to end in `.json`? Do directories have to have a trailing `/`?

Author: If a dirPath is given it will read/create a file `options-per-service.json` in that dir. If a filePath is given it read/create a file with a given name.

Also, the properties in the configuration have changed:
- The option `serviceName` is removed.
A value for the documentation header is derived from the directory name.
- The `npmPackageName` is renamed to `packageName` to align with the OpenApi generator.
- The keys change to the relative paths of the service specifications.
In case you have problems finding the relative path, rerun the generator which will add new default entries to the exising options.

Here is an example how the options change:

```json
//Old Format
{
  "API-TEST_SRV": {
    "directoryName": "test-service",
    "basePath": "/odata/test-service",
    "npmPackageName": "npm-package-test-service",
    "serviceName": "some-text-used-in-documentation-service"
  }
}

//New Format
{

>   "../test-resources/odata-spec/API-TEST_SRV.edmx": {

Reviewer: this path is relative to what? the package.json?

Author: we use `process.cwd()` which means relative to the location the generator is executed from. We could also make it relative to the location of the options-per-serivice so the file location where the path is given. I have no strong opinion on it.

Reviewer: thinking about it I guess making it relative to options per service might be more stable.. unless one moves that file..

I'm not sure. We can also leave it like it is..

    "packageName": "npm-package-test-service",
    "directoryName": "test-service",
    "basePath": "/odata/test-service"
  }
}
```

Note that the default values for the `directoryName` and `packageName` have changed.
They are both based on the directory name which is based on the name of the service specification.

> If you are not happy with the default name use the `optionsPerService` to specify a different value. 

Reviewer: I have trouble understanding this sentence. I thought we are discussing 'optionsPerService' here.

Questions:

- What is the flow (i.e. should I let the generator generate the file and then edit names I don't like?)
-

Author: I tried to make it more clear have a look.
